components/gitpod-db/src/typeorm/user-db-impl.ts

```diff
@@ -395,6 +402,48 @@ export class TypeORMUserDBImpl implements UserDB {
         await repo.save(envVar);
     }
 
+    public async hasSSHPublicKey(userId: string): Promise<boolean> {
+        const repo = await this.getSSHPublicKeyRepo();
+        return !!(await repo.findOne({ where: { userId, deleted: false } }));
+    }
+
+    public async getSSHPublicKeys(userId: string): Promise<UserSSHPublicKey[]> {
+        const repo = await this.getSSHPublicKeyRepo();
+        return repo.find({ where: { userId, deleted: false }, order: { creationTime: "ASC" } });
+    }
+
+    public async addSSHPublicKey(userId: string, value: SSHPublicKeyValue): Promise<UserSSHPublicKey> {
+        const repo = await this.getSSHPublicKeyRepo();
+        const fingerprint = SSHPublicKeyValue.getFingerprint(value);
+        const allKeys = await repo.find({ where: { userId } });
+        const prevOne = allKeys.find((e) => e.fingerprint === fingerprint);
+        if (!!prevOne) {
+            if (prevOne.deleted === false) {
+                throw new Error(`Duplicate public key with ${prevOne.name}`);
+            }
+            prevOne.deleted = false;
+            return repo.save(prevOne);
```

Comment:
Not sure if this will affect `db-sync` or not

Reply:
Any suggestion? Should we delete the UNIQUE INDEX for (userId and fingerprint) in DB and use DB transaction here? 
cc @jankeromnes

Reply:
Hi, thanks for the ping @mustard-mh!

Oh, re-reading your question, I've just now understood the relationship between the UNIQUE INDEX (`UNIQUE KEY ind_unique_userId_fingerprint (userId, fingerprint)` in the migration) and why you're trying to handle rows with `deleted: true` here -- you're worried that, if someone adds a key with a fingerprint of a just-deleted key (but not yet fully removed), MySQL will throw an error because of the duplicate userId/fingerprint, right?

In that case, yes, I think removing the UNIQUE INDEX makes sense, as it allows you to delete & recreate keys without having to wait for full removal of deleted rows (or trying to "undeleted" keys that are in the process of being deleted).

Now, I just don't fully understand why a DB transaction would be useful here. 👀 Could you please explain?